demos/spectre_v1_btb_ca.cc

```diff
@@ -0,0 +1,199 @@
+/*
+ * Copyright 2019 Google LLC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *   https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#ifndef __linux__
+#  error Unsupported OS. Linux required.
+#endif
+
+#include <array>
+#include <cstring>
+#include <iostream>
+
+#include <sched.h>
+#include <signal.h>
+#include <sys/types.h>
+#include <unistd.h>
+
+#include "cache_sidechannel.h"
+#include "instr.h"
+
+// Objective: given some control over accesses to the *non-secret* string
+// "xxxxxxxxxxxxxx", construct a program that obtains "It's a s3kr3t!!!"
+// without ever accessing it in the C++ execution model, using speculative
+// execution and side channel attacks. The public data is intentionally just
+// xxx, so that there are no collisions with the secret and we don't have to
+// use variable offset.
+const char *public_data = "xxxxxxxxxxxxxxxx";
+const char *private_data = "It's a s3kr3t!!!";
+constexpr size_t kAccessorArrayLength = 4096;
+constexpr size_t kCacheLineSize = 64;
+
+// Used in pointer-arithmetics and control-flow. On the child (attacker) it is
+// 0, on the parent (victim) it is non-zero (it stores the pid of the child).
+pid_t pid;
+// Stores the parent pid. Used to terminate the child.
+pid_t ppid;
+
+// DataAccessor provides an interface to access bytes from either the public or
+// the private storage.
+class DataAccessor {
+ public:
+  virtual char GetDataByte(size_t index) = 0;
+  virtual ~DataAccessor() {};
+};
+
+// PrivateDataAccessor returns private_data. Used only by the child (attacker).
+class PrivateDataAccessor: public DataAccessor {
+ public:
+  char GetDataByte(size_t index) override {
+    return private_data[index];
+  }
+};
+
+// PublicDataAccessor returns public_data. Used only by the victim (parent).
+class PublicDataAccessor: public DataAccessor {
```

Comment:
```suggestion
class PublicDataAccessor : public DataAccessor {
```

consider running `clang-format`